apps/condo/domains/organization/integrations/sbbol/sync/requestTransactions.js

```diff
@@ -0,0 +1,170 @@
+ *  @param {String} organizationId
+ *  @param {String} bankIntegrationContextId
+ */
+async function _requestTransactions ({ userId, bankAccounts, context, statementDate, organizationId, bankIntegrationContextId }) {
+    const fintechApi = await initSbbolFintechApi(userId)
+
+    if (!fintechApi) return
+
+    const transactions = []
+
+    for (const bankAccount of bankAccounts) {
+        let page = 1,
+            doRequest = true,
+            timeout = 1000,
+            failedReq = 0
+
+        do {
+            let response = await fintechApi.getStatementTransactions(bankAccount.number, statementDate, page)
+
+            while (!isTransactionsReceived(response)) {
+                if (failedReq > 5) {
+                    break
+                }
+                await new Promise( (resolve) => {
+                    setTimeout(async () => {
+                        response = await fintechApi.getStatementTransactions(
+                            bankAccount.number,
+                            statementDate,
+                            page
+                        )
+                        resolve()
+                    }, timeout)
+                })
+                failedReq++
+                timeout *= 2
+            }
+
+            get(response, 'data.transactions').map( transaction => transactions.push(transaction))
+            page++
+
+            // Checking that the response contains a link to the next page, if it is not there, then all transactions have been received
+            if (isEmpty(get(response, 'data._links', []).filter(link => link.rel === 'next'))) {
+                doRequest = false
+            }
+
+            // WORKFLOW_FAULT means that the request parameters are not valid. Occurs in two cases,
+            // if the report is requested in the future tense and
+            // if the report page does not exist. For example, the total report contains 3 pages, query 4 will return WORKFLOW_FAULT.
+            if (get(transactions, 'error.cause') === 'WORKFLOW_FAULT') doRequest = false
+        } while ( doRequest )
+
+        for (const transaction of transactions) {
+            // If SBBOL returned a transaction with an unsupported currency, do not process
+            if (!isEmpty(ISO_CODES.map( currencyName => get(transaction, 'amount.currencyName') === currencyName))) {
+                const formatedOperationDate = dayjs(transaction.operationDate).format('YYYY-MM-DD')
+                const whereConditions = {
+                    number: transaction.number,
+                    date:  formatedOperationDate,
+                    amount: transaction.amount.amount,
+                    currencyCode: transaction.amount.currencyName,
+                    purpose: transaction.paymentPurpose,
+                    dateWithdrawed: transaction.direction === 'CREDIT' ? formatedOperationDate : null,
+                    dateReceived: transaction.direction === 'DEBIT' ? formatedOperationDate : null,
```

Comment:
I've deleted these fields in my PR #2870

```suggestion
```

Reply:
Yep, I will update it when I transfer it to a banking repository.